core/feed_protocol/annotator/circulation.py

```diff
@@ -0,0 +1,1559 @@
+            kwargs.update(dict(list(facets.items())))
+        if pagination:
+            kwargs.update(dict(list(pagination.items())))
+        return self.url_for(
+            "lane_search",
+            lane_identifier=lane_identifier,
+            library_short_name=self.library.short_name,
+            _external=True,
+            **kwargs,
+        )
+
+    def group_uri(
+        self, work: Work, license_pool: Optional[LicensePool], identifier: Identifier
+    ) -> Tuple[Optional[str], str]:
+        if not work in self.lanes_by_work:
+            return None, ""
+
+        lanes = self.lanes_by_work[work]
+        if not lanes:
+            # I don't think this should ever happen?
+            lane_name = None
+            url = self.url_for(
+                "acquisition_groups",
+                lane_identifier=None,
+                library_short_name=self.library.short_name,
+                _external=True,
+            )
+            title = "All Books"
+            return url, title
+
+        lane = lanes[0]
+        self.lanes_by_work[work] = lanes[1:]
+        lane_name = ""
+        show_feed = False
+
+        if isinstance(lane, dict):
+            show_feed = lane.get("link_to_list_feed", show_feed)
+            title = lane.get("label", lane_name)
+            lane = lane["lane"]
+
+        if isinstance(lane, str):
+            return lane, lane_name
+
+        if hasattr(lane, "display_name") and not title:
+            title = lane.display_name
+
+        if show_feed:
+            return self.feed_url(lane, self.facets), title
+
+        return self.lane_url(lane, self.facets), title
+
+    def lane_url(
+        self, lane: Optional[WorkList], facets: Optional[FacetsWithEntryPoint] = None
+    ) -> str:
+        # If the lane has sublanes, the URL identifying the group will
+        # take the user to another set of groups for the
+        # sublanes. Otherwise it will take the user to a list of the
+        # books in the lane by author.
+
+        if lane and isinstance(lane, Lane) and lane.sublanes:
+            url = self.groups_url(lane, facets=facets)
+        elif lane and (isinstance(lane, Lane) or isinstance(lane, DynamicLane)):
+            url = self.feed_url(lane, facets)
+        else:
+            # This lane isn't part of our lane hierarchy. It's probably
+            # a WorkList created to represent the top-level. Use the top-level
+            # url for it.
+            url = self.default_lane_url(facets=facets)
+        return url
+
+    def annotate_work_entry(
+        self, entry: WorkEntry, updated: Optional[datetime.datetime] = None
+    ) -> None:
+        super().annotate_work_entry(entry, updated=updated)
+
+        if not entry.computed:
+            return
+
+        work = entry.work
+        identifier = entry.identifier or work.presentation_edition.primary_identifier
+
+        permalink_uri, permalink_type = self.permalink_for(identifier)
+        # TODO: Do not force OPDS types
```

Comment:
And here

Reply:
Until we have all endpoints under OPDS 2.0 maybe we should keep the APIs as OPDS1.x and keep the TODO as a reminder.

Reply:
I see now what the todo is mentioning, it wasn't clear to me before. Can you maybe elaborate on the comment a bit more, so our future selves know what we are talking about here?